org.osgi.service.feature/src/org/osgi/service/feature/FeatureService.java

```diff
@@ -33,21 +33,67 @@
      * Get a factory which can be used to build feature model entities.
      * @return A builder factory.
      */
-    public BuilderFactory getBuilderFactory();
+	BuilderFactory getBuilderFactory();
+
+	/**
+	 * Obtain an ID from a Maven ID formatted string. The supported syntax is as
+	 * follows:
+	 * <p>
+	 * {@code groupId ':' artifactId ( ':' type ( ':' classifier )? )? ':' version }
+	 * 
+	 * @param mavenID The Maven ID.
+	 * @return the ID.
+	 */
+	ID getIDfromMavenID(String mavenID);
```

Comment:
Is mavenId a common word? I only used Maven Coordinates in the past.

Reply:
You're right. I always called these 'Maven IDs' but looking at https://maven.apache.org/pom.html#Maven_Coordinates coordinates seems the right term. I changed the parameter name in the latest squashed commit.